libexec/bats

```diff
@@ -55,7 +60,7 @@ expand_path() {
 BATS_LIBEXEC="$(abs_dirname "$0")"
 export BATS_PREFIX="$(abs_dirname "$BATS_LIBEXEC")"
 export BATS_CWD="$(abs_dirname .)"
-export PATH="$BATS_LIBEXEC:$PATH"
+export PATH="$BATS_LIBEXEC:$(getconf PATH 2>/dev/null):$PATH"
```

Comment:
I'm not comfortable modifying the path like this. If you don't already have the POSIX tools first in your path when you run bats, presumably you don't want them there during your test runs, either. 

Do we only need this for sed? If so, let's just define a `bats_sed` function in `bats-exec-test` that invokes the right executable on Solaris.